test/suites/beta/integration/networkinterfaces_test.go

```diff
@@ -0,0 +1,96 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/apis/v1beta1"
+)
+
+var _ = Describe("NetworkInterfaces", func() {
+	BeforeEach(func() {
+		// Ensure that nodes schedule to private subnets. If a node without a public IP is assigned to a public subnet,
+		// and that subnet does not contain any private endpoints to the cluster, the node will be unable to join the
+		// cluster.
+		nodeClass.Spec.SubnetSelectorTerms = []v1beta1.SubnetSelectorTerm{
+			{
+				Tags: map[string]string{
+					"Name":                   "*Private*",
+					"karpenter.sh/discovery": env.ClusterName,
+				},
+			},
+		}
+	})
+
+	DescribeTable(
+		"should correctly create NetworkInterfaces",
+		func(interfaces ...*v1beta1.NetworkInterface) {
+			nodeClass.Spec.NetworkInterfaces = interfaces
+			pod := test.Pod()
+			env.ExpectCreated(pod, nodeClass, nodePool)
+			env.EventuallyExpectHealthy(pod)
+			env.ExpectCreatedNodeCount("==", 1)
+			instance := env.GetInstance(pod.Spec.NodeName)
+			for _, interfaceSpec := range interfaces {
```

Comment:
I think that we have a function that grabs the interfaces from the instance? Can we leverage this function rather than grabbing the instance and doing things with checking the fields?

Reply:
I'll take a look 👍